### PR TITLE
security: update libcurl from 7.71.1 to 7.76.0

### PR DESCRIPTION
**Pull request by Totktonada**
The reason of the update is to protect us against possible MITM attack
from a malicious HTTPS proxy server with trusted certificate when TLS
1.3 is used ([CVE-2021-22890][1]). libcurl versions prior to 7.76.0 can
skip a TLS handshake with a target host in this circumstances.

Other vulnerabilities fixed in the (7.71.1; 7.76.0] version range do not
look relevant to our built-in http client. See [here][2] for the full list.

The CMake version requirement is updated from 3.1 to 3.2, because curl's
CMakeLists.txt has the following clause at beginning:

```cmake
cmake_minimum_required(VERSION 3.2...3.16 FATAL_ERROR)
```

(It was there in vanilla curl 7.71.1 too and we had to remove it in
order to support CMake 2. Now we don't support CMake 2, so it is good
time to get rid of the extra patch upward vanilla curl repository.)

According to the CMake versions table in 8a7702b ('github-ci: purge Debian
Jessie from CI'), CMake 3.2+ is available on all supported OSes.

[1]: https://curl.se/docs/CVE-2021-22890.html
[2]: https://curl.se/docs/vulnerabilities.html

@TarantoolBot document
Title: Now we require CMake 3.2 to build tarantool

In https://github.com/tarantool/doc/issues/1780 we requested
to update the CMake minimum version to 3.1. Now it is time for 3.2. See details in
the linked commit.

Please, update the 'Building from source' manual.

### CODE REVIEW

**Comment by Totktonada** [comment]
Cherry-picking of this commit on 1.10 will conflict on the curl submodule, because the submodule is fixed on different commits on master and 1.10. Just update it to the same commit as on master (tag: `curl-7_76_0`, commit: `3266b35bbe21c68dea0dc7ccd991eb028e6d360c`).

**Review by avtikhon — APPROVED** [review]
Needed commit in-real, LGTM.

**Review by igormunkin — APPROVED** [review]
I have no idea, how to review such a [huge changeset](https://github.com/tarantool/curl/compare/12af024bc85606b14ffc415413a7e86e6bbee7eb...3266b35bbe21c68dea0dc7ccd991eb028e6d360c) in curl. Honestly, I have no idea whether I even need to do this. Regarding Tarantool-related changes everything is fine (even for 1.10 considering b990d36), so the patch is LGTM.

**Comment by Totktonada** [comment]
I made several tests that aim to catch possible memory leak.

TL;DR: The brief testing does not reveal any leak.

1. Simple test with ASAN.

HTTP server: python -m http.server 8080 (in an empty directory)
Build command: CC=clang-11 CXX=clang++-11 cmake . -DCMAKE_BUILD_TYPE=Debug -DENABLE_BACKTRACE=ON -DENABLE_DIST=ON -DENABLE_FEEDBACK_DAEMON=OFF -DENABLE_ASAN=ON && make -j

I built master and the patched version with ASAN and run the following workload:

```
tarantool> httpc = require('http.client')
tarantool> for i = 1, 1000 do httpc.get('http://localhost:8080') end
tarantool> Ctrl+D
```

There are no significant differences (just something from initialization and _rl_malloc).

Increased the workload on the patched version:

```
tarantool> httpc = require('http.client')
tarantool> for i = 1, 10000 do httpc.get('http://localhost:8080') end
tarantool> Ctrl+D
```

There are no significant differences, again.

2. Long test with HTTPS.

Build command: cmake . -DCMAKE_BUILD_TYPE=RelWithDebInfo -DENABLE_BACKTRACE=ON -DENABLE_DIST=ON -DENABLE_FEEDBACK_DAEMON=OFF && make -j

Run the following workload:

```
tarantool> httpc = require('http.client')
tarantool> i=0 while true do assert(httpc.get('https://google.com').status == 200) i = i + 1 print('Done: ' .. i) end
-- Google starts to discard requests after ~500 ones, so continue with my host.
tarantool> i=0 while true do assert(httpc.get('https://tkn.me').status == 200) i = i + 1 print('Done: ' .. i) end
-- ~12000 requests.
-- ~47 minutes overall.
```

Track RSS:

```
$ while true; do echo "[$(date)] $(ps ax -o pid,rss,comm | grep [t]arantool)" >>res.txt; sleep 1; done
```

Results:

After ~6 minutes RSS stabilized on 44532 KiB and didn't changed anymore.

**Comment by Totktonada** [comment]
@igormunkin I hope that being on track with libcurl updates in the better option than trying to support a downstream: this way we'll get bugfixes and security fixes without paying constant attention to libcurl's changes (just check for security fixes periodically). Fortunately, the API is quite stable.

**Comment by olegrok** [comment]
Seems this patch was accidentally reverted by https://github.com/tarantool/tarantool/commit/660686e3f95ee210f421157c4534107d8eb75c6c#diff-1dfc812e3f3840f507f82649f056243ca4b656224c4ccf36ab4fa8fed2e7be13

**Comment by Korablev77** [comment]
Sorry guys, reverted back in https://github.com/tarantool/tarantool/commit/9610c7410ce210cb29de0a616f357c1a17392047. Thanks for reporting.

**Comment by Totktonada** [comment]
The documentation issue: https://github.com/tarantool/doc/issues/2065
